pkg/plugins/resources/memory/store.go

```diff
@@ -158,6 +158,9 @@ func (c *memoryStore) Get(_ context.Context, r model.Resource, fs ...store.GetOp
 	if record == nil {
 		return store.ErrorResourceNotFound(r.GetType(), opts.Name, opts.Mesh)
 	}
+	if opts.Version != "" && opts.Version != record.Version.String() {
+		return store.ErrorResourceNotFound(r.GetType(), opts.Name, opts.Mesh)
```

Comment:
`ErrorResourceConflict` (409) or `ErrorResourcePreconditionFailed` (412)  might be a better fit (resource is there, but its state is different from what is expected)

Reply:
went with 412